L1Trigger/L1CaloTrigger/plugins/L1CaloJetHTTProducer.cc

```diff
@@ -0,0 +1,123 @@
+
+#include <iostream>
+
+// Run2/PhaseI output formats
+#include "DataFormats/L1Trigger/interface/L1JetParticleFwd.h"
+#include "DataFormats/L1Trigger/interface/Jet.h"
+// GenJets if needed
+#include "DataFormats/JetReco/interface/GenJet.h"
+#include "DataFormats/JetReco/interface/GenJetCollection.h"
+
+class L1CaloJetHTTProducer : public edm::EDProducer {
+public:
+  explicit L1CaloJetHTTProducer(const edm::ParameterSet&);
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+  double EtaMax;
+  double PtMin;
+
+  edm::EDGetTokenT<BXVector<l1t::Jet>> bxvCaloJetsToken_;
+  edm::Handle<BXVector<l1t::Jet>> bxvCaloJetsHandle;
+
+  // Gen jet collections are only loaded and used if requested
+  // (use_gen_jets == true)
+  edm::EDGetTokenT<std::vector<reco::GenJet>> genJetsToken_;
+  edm::Handle<std::vector<reco::GenJet>> genJetsHandle;
+
+  bool debug;
+
+  bool use_gen_jets;
+};
+
+L1CaloJetHTTProducer::L1CaloJetHTTProducer(const edm::ParameterSet& iConfig)
+    : EtaMax(iConfig.getParameter<double>("EtaMax")),
+      PtMin(iConfig.getParameter<double>("PtMin")),
+      bxvCaloJetsToken_(consumes<BXVector<l1t::Jet>>(iConfig.getParameter<edm::InputTag>("BXVCaloJetsInputTag"))),
+      genJetsToken_(consumes<std::vector<reco::GenJet>>(iConfig.getParameter<edm::InputTag>("genJets"))),
+      debug(iConfig.getParameter<bool>("debug")),
+      use_gen_jets(iConfig.getParameter<bool>("use_gen_jets"))
+
+{
+  produces<float>("CaloJetHTT");
+}
+
+void L1CaloJetHTTProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  // Output collections
+  std::unique_ptr<float> CaloJetHTT(new float);
+
+  *CaloJetHTT = 0.;
+
+  // CaloJet HTT for L1 collections
+  if (!use_gen_jets) {
+    iEvent.getByToken(bxvCaloJetsToken_, bxvCaloJetsHandle);
+
+    if (bxvCaloJetsHandle.isValid()) {
+      for (const auto& caloJet : *bxvCaloJetsHandle.product()) {
+        if (caloJet.pt() < PtMin)
+          continue;
+        if (fabs(caloJet.eta()) > EtaMax)
+          continue;
+        *CaloJetHTT += float(caloJet.pt());
+      }
+    }
+
+    if (debug) {
+      LogDebug("L1CaloJetHTTProducer") << " BXV L1CaloJetCollection JetHTT = " << *CaloJetHTT << " for PtMin " << PtMin
+                                       << " and EtaMax " << EtaMax << "\n";
+    }
+  }
+
+  // CaloJet HTT for gen jets
+  if (use_gen_jets) {
+    iEvent.getByToken(genJetsToken_, genJetsHandle);
+
+    if (genJetsHandle.isValid()) {
+      for (const auto& genJet : *genJetsHandle.product()) {
+        if (genJet.pt() < PtMin)
+          continue;
+        if (fabs(genJet.eta()) > EtaMax)
+          continue;
+        *CaloJetHTT += float(genJet.pt());
+      }
+    }
+
+    if (debug) {
+      LogDebug("L1CaloJetHTTProducer") << " Gen Jets HTT = " << *CaloJetHTT << " for PtMin " << PtMin << " and EtaMax "
+                                       << EtaMax << "\n";
+    }
+  }
+
+  iEvent.put(std::move(CaloJetHTT), "CaloJetHTT");
```

Comment:
Do I understand correctly that, if genJet collection is not available but it is set to use, this module still RUN (no crash expected)?

Reply:
I would assume so, but I have not tested with files where no genJet collection is available but with use_gen_jets set to True.